test/unit/init/AppRunner.test.ts

```diff
@@ -290,6 +290,38 @@ describe('AppRunner', (): void => {
       expect(app.start).toHaveBeenLastCalledWith();
     });
 
+    it('runs the server honoring env variables.', async(): Promise<void> => {
+      const { env } = process;
+      // Set logging level to debug
+      env.CSS_LOGGING_LEVEL = 'debug';
+
+      await expect(new AppRunner().runCli([ 'node', 'script' ])).resolves.toBeUndefined();
+
+      expect(ComponentsManager.build).toHaveBeenCalledTimes(1);
+      // Check logLevel to be set to debug instead of default `info`
+      expect(ComponentsManager.build).toHaveBeenCalledWith({
+        dumpErrorState: true,
+        logLevel: 'debug',
+        mainModulePath: joinFilePath(__dirname, '../../../'),
+      });
+      expect(manager.configRegistry.register).toHaveBeenCalledTimes(1);
+      expect(manager.configRegistry.register)
+        .toHaveBeenCalledWith(joinFilePath(__dirname, '/../../../config/default.json'));
+      expect(manager.instantiate).toHaveBeenCalledTimes(2);
+      expect(manager.instantiate).toHaveBeenNthCalledWith(1, 'urn:solid-server-app-setup:default:CliResolver', {});
+      expect(cliExtractor.handleSafe).toHaveBeenCalledTimes(1);
+      expect(cliExtractor.handleSafe).toHaveBeenCalledWith([ 'node', 'script' ]);
+      expect(settingsResolver.handleSafe).toHaveBeenCalledTimes(1);
+      expect(settingsResolver.handleSafe).toHaveBeenCalledWith(defaultParameters);
+      expect(manager.instantiate).toHaveBeenNthCalledWith(2,
+        'urn:solid-server:default:App',
+        { variables: defaultVariables });
+      expect(app.start).toHaveBeenCalledTimes(1);
+      expect(app.start).toHaveBeenLastCalledWith();
+
+      delete env.CSS_LOGGING_LEVEL;
```

Comment:
This is very nitpicky (but that's always fun right?) but this should be `process.env = env`. Because if originally there was a `CSS_LOGGING_LEVEL` in `env` before this test it would be removed afterwards.

Reply:
This is not the comment I wanted to put here, I editted it and forgot to save, give me a sec. 😄

Reply:
So what I wanted to say is that we should make sure to copy the env object like in the yargs test:

https://github.com/CommunitySolidServer/CommunitySolidServer/blob/df2926a52ff7ca45a4a2b19871f293b791360c05/test/unit/init/cli/YargsCliExtractor.test.ts#L79-L81
https://github.com/CommunitySolidServer/CommunitySolidServer/blob/df2926a52ff7ca45a4a2b19871f293b791360c05/test/unit/init/cli/YargsCliExtractor.test.ts#L87

Unless this doesn't work because of the big comment I put in that test? But I sort of forgot what I meant there exactly.